src/processes/helper/userIdsToProcess.js

```diff
@@ -23,87 +42,88 @@ const findF2UserIds = async (userId) => {
   const f2UserIds = await UserFollowUser.findAll({
     where: {
       [Op.and]: [
-        { user_id_follower: { [Op.in]:followedUserIds} },
-        { user_id_followed: { [Op.notIn]: excludeUserIds } }
+        {user_id_follower: {[Op.in]: followedUserIds}},
+        {user_id_followed: {[Op.notIn]: excludeUserIds}}
       ]
     },
     limit: 1000,
     attributes: ['user_id_followed'],
-    raw : true
-  }).then( users => users.map(user => user.user_id_followed));
+    raw: true
+  }).then((users) => users.map((user) => user.user_id_followed));
 
-  return f2UserIds
-}
-
-const findBlockedUserIds = async (userId) => {
-  const blockedUserIds = await UserBlockedUser.findAll({
-    where: {
-      user_id_blocker: userId,
-    },
-    limit: 1000,
-    attributes: ['user_id_blocked'],
-    raw : true
-  }).then( users => users.map(user => user.user_id_blocked));
-
-  return blockedUserIds
-}
+  return f2UserIds;
+};
 
 const findRelatedUserIds = async (userId) => {
   const [followedByUserId, f2UserIds, blockedUserIds] = await Promise.all([
     findFollowingUserIds(userId),
     findF2UserIds(userId),
-    findBlockedUserIds(userId),
+    findBlockedUserIds(userId)
   ]);
 
-  let listUser = [...followedByUserId, ...f2UserIds, ...blockedUserIds];
-  let listUniqueUser = [...new Set(listUser)]
-  return listUniqueUser
-}
+  const listUser = [...followedByUserId, ...f2UserIds, ...blockedUserIds];
+  const listUniqueUser = [...new Set(listUser)];
+  return listUniqueUser;
+};
 
-const findUnrelatedUserIds = async (userId) => {
-  const relatedUserIds = await findRelatedUserIds(userId)
+const findUnrelatedUserIds = async (relatedUserIds) => {
   const unrelatedUserIds = await User.findAll({
     where: {
-      user_id: { [Op.notIn]: relatedUserIds }
+      user_id: {[Op.notIn]: relatedUserIds},
+      is_anonymous: false
     },
     limit: 1000,
     attributes: ['user_id'],
-    raw : true
-  }).then( users => users.map(user => user.user_id));
+    raw: true
+  }).then((users) => users.map((user) => user.user_id));
 
-  return unrelatedUserIds
-}
+  return unrelatedUserIds;
+};
+
+const findUnrelatedAnonUserIds = async (relatedUserIds) => {
+  const unrelatedUserIds = await User.findAll({
+    where: {
+      user_id: {[Op.notIn]: relatedUserIds},
+      is_anonymous: true
+    },
+    limit: 1000,
+    attributes: ['user_id'],
+    raw: true
+  }).then((users) => users.map((user) => user.user_id));
+
+  return unrelatedUserIds;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Again, the limit of 1000 might not be sufficient if there are more than 1000 anonymous users in the database that match the criteria. Consider removing this limit or increasing it.



<!-- This is an auto-generated comment by OSS CodeRabbit -->